lib/segment/src/index/field_index/geo_index.rs

```diff
@@ -526,43 +526,46 @@ impl PayloadFieldIndex for GeoMapIndex {
         if let Some(geo_bounding_box) = &condition.geo_bounding_box {
             let geo_hashes = rectangle_hashes(geo_bounding_box, GEO_QUERY_MAX_REGION);
             let geo_condition_copy = geo_bounding_box.clone();
-            return Some(Box::new(self.get_iterator(geo_hashes).filter(
-                move |point| {
-                    self.point_to_values
-                        .get(*point as usize)
-                        .unwrap()
-                        .iter()
-                        .any(|point| geo_condition_copy.check_point(point.lon, point.lat))
-                },
-            )));
+            return Some(Box::new(
+                self.get_iterator(geo_hashes.unwrap_or(vec![]))
```

Comment:
agree,  I was not 100% sure when I made an unwrap. I can consider using `match` here if you think that will work. I'm only in the beginning of my 🦀 journey, so open to suggestions.

Reply:
You could probably return early with an error, like:

```diff
-let geo_hashes = rectangle_hashes(geo_bounding_box, GEO_QUERY_MAX_REGION);
+let geo_hashes = rectangle_hashes(geo_bounding_box, GEO_QUERY_MAX_REGION)?;
```

That way you can remove the unwrap from `geo_hashes` further down the line.

Reply:
It is probably better to change the function signature to return an `Result` rather than an `Option`.

Then using `?` without `.ok()` should work fine.

Reply:
Got it, let me adjust that